index.bs

```diff
@@ -1114,13 +1114,14 @@ To <dfn>parse aggregatable trigger data</dfn> given an [=ordered map=] |map|:
     1. If |value|["`key_piece`"] does not [=map/exist=] or is not a [=string=], return null.
     1. Let |keyPiece| be the result of running [=parse an aggregation key piece=] with |value|["`key_piece`"].
     1. If |keyPiece| is an error, return null.
-    1. If |value|["`source_keys`"] does not [=map/exist=] or is not a [=list=], return null.
-    1. If |value|["`source_keys`"]'s [=list/size=] is greater than the user agent's
-        [=max aggregation keys per attribution=], return null.
-    1. Let |sourceKeys| be a new [=set=].
-    1. [=list/iterate|For each=] |sourceKey| of |value|["`source_keys`"]:
-        1. If |sourceKey| is not a [=string=], return null.
-        1. [=set/Append=] |sourceKey| to |sourceKeys|.
+    1. Let |sourceKeys| be a new [=ordered set=].
+    1. If |value|["`source_keys`"] [=map/exists=]:
+        1. If |value|["`source_keys`"] is not a [=list=], return null.
+        1. If |value|["`source_keys`"]'s [=list/size=] is greater than the user agent's
+            [=max aggregation keys per attribution=], return null.
+        1. [=list/iterate|For each=] |sourceKey| of |value|["`source_keys`"]:
+            1. If |sourceKey| is not a [=string=], return null.
+            1. [=set/Append=] |sourceKey| to |sourceKeys|.
```

Comment:
Just checking: Are we also supposed to be validating the length of these keys?

Reply:
Similar to filters string, we need to decide whether to validate that in the spec.